sshd-common/src/main/java/org/apache/sshd/common/keyprovider/KeyPairProvider.java

```diff
@@ -83,8 +93,8 @@ public KeyPair loadKey(SessionContext session, String type) {
             }
 
             @Override
-            public Iterable<String> getKeyTypes(SessionContext session) {
-                return Collections.emptyList();
+            public Set<String> getKeyTypes(SessionContext session) {
```

Comment:
Why change this ? I see no reason... - please revert the change or provide compelling reason as to why this is 100% necessary. Please note that some code may rely on `Iterable`...

Reply:
Hehe, I was pretty sure you would say this :) I don't have a reason except avoiding some back and forth converting, will change it back to `Iterable` again.